app/models/manageiq/providers/ibm_cloud/container_manager.rb

```diff
@@ -0,0 +1,201 @@
+class ManageIQ::Providers::IbmCloud::ContainerManager < ManageIQ::Providers::Kubernetes::ContainerManager
+  require_nested :Container
+  require_nested :ContainerGroup
+  require_nested :ContainerNode
+  require_nested :EventCatcher
+  require_nested :EventParser
+  require_nested :Refresher
+  require_nested :RefreshWorker
+
+  def self.ems_type
+    @ems_type ||= "iks".freeze
+  end
+
+  def self.description
+    @description ||= "IBM Cloud Kubernetes Service".freeze
+  end
+
+  def self.display_name(number = 1)
+    n_('Container Provider (IBM Cloud)', 'Container Providers (IBM Cloud)', number)
+  end
+
+  def self.default_port
+    443
+  end
+
+  def connect_options(options = {})
+    authentication = authentication_best_fit(options.fetch(:auth_type, "bearer"))
+    token = self.class.get_token(authentication.auth_key).id_token
+
+    super.merge(
+      :bearer => token,
+      :region => provider_region
+    )
+  rescue IbmCloudIam::ApiError => err
+    error_message = JSON.parse(err.response_body)["message"]
+    _log.error("IAM authentication failed: #{err.code} #{error_message}")
+  end
+
+  def self.kubernetes_auth_options(options)
+    {
+      :bearer_token => options[:bearer]
+    }
+  end
+
+  def self.verify_credentials(args)
+    default_endpoint = args.dig("endpoints", "default")
+    hostname, port, security_protocol, certificate_authority = default_endpoint&.values_at(
+      "hostname", "port", "security_protocol", "certificate_authority"
+    )
+    region = args["provider_region"]
+
+    api_key = ManageIQ::Password.try_decrypt(args.dig("authentications", "bearer", "auth_key"))
+    token = get_token(api_key).id_token
+
+    unless certificate_authority.nil?
+      cert_store = OpenSSL::X509::Store.new
+                                       .add_cert(
+                                         OpenSSL::X509::Certificate.new(certificate_authority)
+                                       )
+    end
+
+    options = {
+      :bearer      => token,
+      :region      => region,
+      :ssl_options => {
+        :verify_ssl => security_protocol == 'ssl-without-validation' ? OpenSSL::SSL::VERIFY_NONE : OpenSSL::SSL::VERIFY_PEER,
+        :cert_store => cert_store
```

Comment:
Did `:ssl_options => {:certificate_authority => certificate_authority}` not work here?

Reply:
I was unable to get this to work, seems like the cert is dumped as a string:
```
[----] E, [2021-09-13T15:29:08.799616 #41510:93a8] ERROR -- evm: MIQ(MiqQueue#deliver) Message id: [4243], Error: [HTTP status code , SSL_connect returned=1 errno=0 state=error: certificate verify failed (self signed certificate in certificate chain)]
[----] E, [2021-09-13T15:29:08.799734 #41510:93a8] ERROR -- evm: [Kubeclient::HttpError]: SSL_connect returned=1 errno=0 state=error: certificate verify failed (self signed certificate in certificate chain)  Method:[block (2 levels) in <class:LogProxy>]
[----] E, [2021-09-13T15:29:08.799818 #41510:93a8] ERROR -- evm: /Users/nasarkhan/.rvm/gems/ruby-2.7.3/gems/kubeclient-4.9.2/lib/kubeclient/common.rb:130:in `rescue in handle_exception'
/Users/nasarkhan/.rvm/gems/ruby-2.7.3/gems/kubeclient-4.9.2/lib/kubeclient/common.rb:120:in `handle_exception'
/Users/nasarkhan/.rvm/gems/ruby-2.7.3/gems/kubeclient-4.9.2/lib/kubeclient/common.rb:596:in `fetch_entities'
/Users/nasarkhan/.rvm/gems/ruby-2.7.3/gems/kubeclient-4.9.2/lib/kubeclient/common.rb:583:in `load_entities'
/Users/nasarkhan/.rvm/gems/ruby-2.7.3/gems/kubeclient-4.9.2/lib/kubeclient/common.rb:134:in `discover'
/Users/nasarkhan/Documents/manageiq/plugins/manageiq-providers-kubernetes/app/models/manageiq/providers/kubernetes/container_manager.rb:850:in `kubernetes_connect'
/Users/nasarkhan/Documents/manageiq/plugins/manageiq-providers-kubernetes/app/models/manageiq/providers/kubernetes/container_manager.rb:64:in `raw_connect'
/Users/nasarkhan/Documents/manageiq/plugins/manageiq-providers-ibm_cloud/app/models/manageiq/providers/ibm_cloud/container_manager.rb:58:in `verify_credentials'
```

I found the `cert_store` that I've employed over here: https://github.com/abonas/kubeclient#ssl